code/modules/economy/money_account.dm

```diff
@@ -0,0 +1,227 @@
+	///Unique Account Number Identifier for the account     -Username
+	var/account_number = 0
+	///4 digit pin password for accessing the money account -Password
+	var/account_pin = 0
+	///How many space credits are in the account
+	var/credit_balance
+	///Tracking log of all actions on the account, for IC usage & can have entries deleted
+	var/list/account_log = list()
+	///Tracking log of all actions on the account, used for admin logging and debugging
+	var/list/hidden_account_log = list()
+
+	///reference to parent account database, only used for GC
+	var/datum/money_account_database/database_holder
+
+	///Level of security on the money account
+	var/security_level
+	///Bool - Is this account locked out from being used completely?
+	var/suspended = FALSE
+	///Type of account this is
+	var/account_type = ACCOUNT_TYPE_PERSONAL
+	///the amount this account recieves every payday
+	var/payday_amount = CREW_BASE_PAY_LOW
+
+	///The nanobank programs associated with this account, used for notifying crew members through PDA, this is a lazy list
+	var/list/associated_nanobank_programs
+	///The requests made to this money_account used for ATM and NanoBank credit transfers, this is a lazy list
+	var/list/transfer_requests
+
+	///pay bonuses for the next pay period for this account, this is a lazy list
+	var/list/pay_check_bonuses
+	///pay deductions for the next pay period for this account, this is a lazy list
+	var/list/pay_check_deductions
+
+/datum/money_account/New(_account_name, starting_balance = 0, _security_level = ACCOUNT_SECURITY_ID, _account_type = ACCOUNT_TYPE_PERSONAL, _payday_amount = CREW_BASE_PAY_LOW)
+	account_name = _account_name
+	credit_balance = starting_balance
+	payday_amount = _payday_amount
+	security_level = _security_level
+	account_number = SSeconomy.generate_account_number()
+	account_type = _account_type
+	account_pin = rand(10000, 99999)
+	//update SSeconomy stats
+	SSeconomy.total_space_credits += starting_balance
+
+/datum/money_account/Destroy(force)
+	//we don't need to worry about nanobank programs here because they auto GC themselves
+	QDEL_LIST(account_log)
+	QDEL_LIST(hidden_account_log)
+	if(!QDELETED(database_holder))
+		if(account_type == ACCOUNT_TYPE_PERSONAL)
+			database_holder.user_accounts -= src //remove reference to this account incase this was not deleted the "correct" way through an account db
+		else if(!force && istype(database_holder, /datum/money_account_database/main_station) && account_type == ACCOUNT_TYPE_DEPARTMENT)
+			return QDEL_HINT_LETMELIVE //we do not want department accounts being deleted ever unless we really really mean to
+
+	for(var/datum/transfer_request/request as anything in transfer_requests)
+		resolve_transfer_request(request, FALSE)
+		//while we're here, lets do the GC a favor and get rid of these transfer request datums
+	database_holder = null
+	SSeconomy.total_space_credits -= credit_balance
+	SSeconomy.space_credits_destroyed += credit_balance
+	credit_balance = 0
```

Comment:
```suggestion
```
Primitives like numbers or strings don't need to be unset

Reply:
removed